src/Controller/SearchController.php

```diff
@@ -13,23 +13,44 @@ public function initialize(): void
 
 	public function index()
     {
-        //array_map([$this, 'loadModel'], ['Words']);
         $sitelang = $this->languageinfo();
         $q = trim($this->request->getQuery('q'));
         $displayType = $this->request->getQuery('displayType');
 
+        // Always get the full set for per-language count (not dependent on pagination)
+        $allWords = $this->fetchTable('Words')->find('searchResults', querystring: $q, langid: $sitelang->id)->contain(['Origins'])->toArray();
+        $originCounts = [];
+        foreach ($allWords as $word) {
+            if (!empty($word->origins)) {
+                foreach ($word->origins as $origin) {
+                    $originName = $origin->origin;
+                    if (!isset($originCounts[$originName])) {
+                        $originCounts[$originName] = 0;
+                    }
+                    $originCounts[$originName]++;
+                }
+            }
+        }
+        $originSummary = '';
+        if (!empty($originCounts)) {
+            $parts = [];
+            foreach ($originCounts as $lang => $ocount) {
+                $parts[] = $ocount . ' were from ' . $lang;
```

Comment:
There shouldn't be any text sent from the controller to the template. Just variables.